core-web/src/main/java/org/silverpeas/core/webapi/upload/ComponentInstanceFileUploadVerification.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2000 - 2016 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception. You should have recieved a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "http://www.silverpeas.org/docs/core/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.silverpeas.core.webapi.upload;
+
+import org.silverpeas.core.admin.component.model.ComponentInstLight;
+import org.silverpeas.core.admin.service.OrganizationController;
+import org.silverpeas.core.io.upload.UploadedFile;
+import org.silverpeas.core.util.ServiceProvider;
+
+import java.io.File;
+import java.util.Optional;
+
+/**
+ * It is a process implied within the manufacturing of a new file upload.
+ * </p>
+ * When a file upload is performed, it is first verified according to its name (the file is not
+ * yet on the server), then is is verified a second time after than the upload is completed
+ * successfully. In some circumstances, according to the application, some verifications have to be
+ * performed in the behalf of the new file upload. The file upload process is unaware of these
+ * circumstances and it cannot know what verification to perform; It is the responsibility of the
+ * application to perform such verifications. This is why an implementation of this interface
+ * qualified by a name that satisfies the following convention <code>[COMPONENT
+ * NAME]InstanceFileUploadVerification</code> is looked for by the file upload process and then
+ * invoked if it is has been found.
+ * </p>
+ * Any application that requires specific actions when a file upload is performed has to implement
+ * this interface and the implementation has to be qualified with the @Named annotation by a name
+ * satisfying the following convention <code>[COMPONENT NAME]FileUploadVerification</code>. For
+ * example, for an application Kmelia, the implementation must be qualified with
+ * <code>@Named("kmeliaInstanceFileUploadVerification") </code>
+ * @author Yohann Chastagnier
+ */
+public interface ComponentInstanceFileUploadVerification {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move constants to a class or enum. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1214)